apps/api/src/app/environments-v1/usecases/output-renderers/expand-email-editor-schema.usecase.ts

```diff
@@ -92,60 +135,74 @@ export class ExpandEmailEditorSchemaUsecase {
-     * element is situated on the container of the list making the list a
-     *  child of the for element, if we iterate it we will get the
-     *  wrong behavior of multiple lists instead of list with multiple items.
-     * due to that when we choose the content to iterate in case we find a list we drill down additional level
-     * and iterate on the list items
-     * this prevents us from
-     * 1. item1
-     * 1. item2
-     *
-     * and turns it into
-     * 1.item1
-     * 2.item2
-     * which is the correct behavior
-     *
-     */
-    if ((this.isOrderedList(templateContent) || this.isBulletList(templateContent)) && templateContent[0].content) {
-      return [{ ...templateContent[0], content: this.regularExpansion(eachObject, templateContent[0].content) }];
-    }
+  /**
+   * For 'each' node, multiply the content by the number of items in the iterable array
+   * and add indexes to the placeholders.
+   *
+   * @example
+   * node:
+   * {
+   *   type: 'each',
+   *   attrs: { each: 'payload.comments' },
+   *   content: [
+   *     { type: 'variable', text: '{{ payload.comments.author }}' }
+   *   ]
+   * }
+   *
+   * variables:
+   * { payload: { comments: [{ author: 'John Doe' }, { author: 'Jane Doe' }] } }
+   *
+   * result:
+   * [
+   *   { type: 'text', text: '{{ payload.comments[0].author }}' },
+   *   { type: 'text', text: '{{ payload.comments[1].author }}' }
+   * ]
+   *
+   */
+  private multiplyForEachNode(
+    node: TipTapNode & { attrs: { each: unknown } },
+    variables: Record<string, unknown>
+  ): TipTapNode[] {
+    const iterablePath = node.attrs.each as string;
+    const nodeContent = node.content || [];
+    const expandedContent: TipTapNode[] = [];
 
-    return this.regularExpansion(eachObject, templateContent);
-  }
+    const iterableArray = this.getValueByPath(variables, iterablePath) as [{ [key: string]: unknown }];
+
+    for (const [index] of iterableArray.entries()) {
+      const contentToExpand =
+        (this.isOrderedList(nodeContent) || this.isBulletList(nodeContent)) && nodeContent[0].content
+          ? nodeContent[0].content
+          : nodeContent;
 
-  private removeNodeFromParent(node: TipTapNode, parentNode?: TipTapNode) {
-    if (parentNode && parentNode.content) {
-      parentNode.content.splice(parentNode.content.indexOf(node), 1);
+      const hydratedContent = this.addIndexesToPlaceholders(contentToExpand, iterablePath, index);
+      expandedContent.push(...hydratedContent);
     }
-  }
 
-  private async hideShowIfNeeded(
-    variables: Record<string, unknown>,
-    node: TipTapNode & { attrs: { [MailyAttrsEnum.SHOW_IF_KEY]: unknown } },
-    parentNode?: TipTapNode
-  ): Promise<void> {
-    const { [MailyAttrsEnum.SHOW_IF_KEY]: showIfKey } = node.attrs;
+    return expandedContent;
+  }
 
-    if (showIfKey === undefined) {
-      return;
-    }
+  private addIndexesToPlaceholders(nodes: TipTapNode[], iterablePath: string, index: number): TipTapNode[] {
```

Comment:
what do you think about?

```suggestion
  private addIndexesToLiquidOutput(nodes: TipTapNode[], iterablePath: string, index: number): TipTapNode[] {
```

Reply:
But the placeholders (`{{payload.something}}`) are not liquid output right? They are more Liquid input, since the parsing by Liquid happens later and this is usecase is an input for that:

```ts
 const parsedTipTap = await this.parseTipTapNodeByLiquid(expandedMailyContent, renderCommand);
 ```

Reply:
Liquid JS calls the value with filters wrapped with `{{...}}` an [Output](https://liquidjs.com/tutorials/intro-to-liquid.html#Outputs)  i using Output instead of placeholder would help us unify and help us to understand each other better.

and the main logic in this function is 
`newNode.text = nodePlaceholder.replace(iterablePath, `${iterablePath}[${index}]`);`
which creates a liquid js Output.
Let me know what you think about it.


In addition, apologies for missing this earlier I noticed that we are using `string.replace`. Instead, we should use `wrapInLiquidOutput` whenever we have a Maily `fallback` attribute.